src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -835,6 +835,10 @@ m2mTest(const std::shared_ptr<xrt_core::device>& _dev, boost::property_tree::ptr
   auto mem_topo = reinterpret_cast<const mem_topology*>(membuf.data());
 
   for (auto& mem : boost::make_iterator_range(mem_topo->m_mem_data, mem_topo->m_mem_data + mem_topo->m_count)) {
+    std::string str((char *)mem.m_tag);
```

Comment:
No C-style cast and don't assume null terminated char* when constructing std::string, use either
`auto c_str = reinterpret_cast<const char*>(mem.m_tag);`
`std::string str{c_str, std::min(strlen(c_str), 16)};  // magic 16 from xclbin.h`
or
`std::string str{c_str, std::find(c_str, c_str + 16, 0)};`  // magic 16 from xclbin.h